chia/wallet/coin_selection.py

```diff
@@ -135,10 +135,11 @@ def select_smallest_coin_over_target(target: uint128, sorted_coin_list: List[Coi
 # we use this to find the set of coins which have total value closest to the target, but at least the target.
 # IMPORTANT: The coins have to be sorted in descending order or else this function will not work.
 def knapsack_coin_algorithm(
-    smaller_coins: List[Coin], target: uint128, max_coin_amount: int, max_num_coins: int
+    smaller_coins: List[Coin], target: uint128, max_coin_amount: int, max_num_coins: int, seed: bytes = b"knapsack seed"
 ) -> Optional[Set[Coin]]:
     best_set_sum = max_coin_amount
     best_set_of_coins: Optional[Set[Coin]] = None
+    random.seed(seed)

```

Comment:
it's a bit of a nitpick, but I think it would be cleaner to create a new `Random` object with this seed, and then ask it for random numbers. With this change, if any other part of the program depends on pseudo random numbers, they won't be so random anymore after calling this function (because you're re-seeding the global state)

Reply:
actually, I think it's quite risky to seed the global PRNG.

Reply:
> actually, I think it's quite risky to seed the global PRNG.

agreed

Reply:
yeah I didn't realize it was global. whoops, that was a huge oversight from my part. I'll do what you suggested